torchvision/datasets/svhn.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import print_function
+import scipy.io as sio
+import torch.utils.data as data
+from PIL import Image
+import os
+import os.path
+import errno
+import numpy as np
+
+class SVHN(data.Dataset):
+    url = ""
+    filename = ""
+    file_md5 = ""
+    
+    def __init__(self, root, dataset='train', transform=None, target_transform=None, download=False):
+        self.root = root
+        self.transform = transform
+        self.target_transform = target_transform
+        self.dataset = dataset  # training set or test set or extra set
+        
+        # download and load the data 
+        if self.dataset=='train':
```

Comment:
Maybe you could condensate these code-paths into a single one?
The only difference that I see is that you change the values of `url`, `filename` and `file_md5` for each set.
Maybe you could instead have a dictionary for each one of those fields, where the key is the set type (`train` / `extra`, `test`), and the values are the different urls, like done in [here](https://github.com/pytorch/vision/pull/63/files#diff-629efaeb461f617e4cdba4ef6b18222eR16)? Then you could have 
```python
self.url = url[dataset]
```